x/cork/client/cli/tx.go

```diff
@@ -16,5 +23,137 @@ func GetTxCmd() *cobra.Command {
 	}
 
 	// todo(mvid): figure out what is useful, implement
-	return oracleTxCmd
+	return corkTxCmd
+}
+
+// GetCmdSubmitAddProposal implements the command to submit a cellar id addition proposal
+func GetCmdSubmitAddProposal() *cobra.Command {
+	bech32PrefixAccAddr := sdk.GetConfig().GetBech32AccountAddrPrefix()
+
+	cmd := &cobra.Command{
+		Use:   "add-cellar-id [proposal-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a cellar id addition proposal",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a cellar addition proposal along with an initial deposit.
+The proposal details must be supplied via a JSON file.
+
+Example:
+$ %s tx gov submit-proposal add-cellar-id <path/to/proposal.json> --from=<key_or_address>
+
+Where proposal.json contains:
+
+{
+  "title": "Dollary-doos LP Cellar Proposal",
+  "description": "I have a hunch",
+  "cellar_ids": ["0x123", "0x456"],
+  "deposit": "1000stake"
+}
+`,
+				bech32PrefixAccAddr,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			proposal := types.AddManagedCellarIDsProposalWithDeposit{}
+			contents, err := ioutil.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			if err = clientCtx.Codec.UnmarshalJSON(contents, &proposal); err != nil {
```

Comment:
It's worth adding a test to make sure you can deserialize the JSON as you expect. I have a vague concern that using CellarIDSet might do something weird rather than a string implementation. Example: https://github.com/PeggyJV/gravity-bridge/blob/main/module/x/gravity/client/cli/tx_test.go

Reply:
Test added